proposal/emjames/proposal.rst

```diff
@@ -0,0 +1,237 @@
+=========================================
+Proposal for Open Data Center Environment
+=========================================
+
+Basic information
+=================
+
+The GitHub repository (URL) hosting the term project:
+https://github.com/emjames/open-data-center-environment
+
+A toolkit for building data center environments and testing them with algorithms to improve data center's energy efficiency.
+
+Problem to solve
+================
+
+This project will have a similar approach to the authors found in [2] and [3] it will also bring ideas from [1]. The main difference is that this project aims to have multiple backends available, as well as making it easier to use, test and modify different learning algorithms (think batteries included).
+
+There exist different approaches for modeling data centers.
+
+.. list-table:: Comparison of existing modeling approaches
+   :widths: 25 50 50
+   :header-rows: 1
+
+
+   * - Modeling approach
+     - Pros
+     - Cons
+   * - | eQuest
+       | EnergyPlus
+       | TRNSYS
+
+     - | Conventional
+       | Existing research
+
+     - | Imperative programming
+       | Math equations are interlaced with numerical solvers (written in source codes, making it difficult to maintain)
+       | Difficult to support fast prototyping
+       | Difficult to implement new control logic
+       | Difficult to perform multi-domain simulation and integration
+
+   * - | Physical models
+       | Equations, physical laws
+
+     - Error of 1 ~ 5%
+
+     - | Require extensive measurements
+       | Static
+       | Extensive assumptions
+
+   * - Data driven models
+
+     - | Don't require known laws
+       | Can speed up simulation
+       | Simulate data center parts with ML algorithms
+     - Need large datasets
+
+
+The overall objective of the Open Data Center Environment project:
+
+Make environments with different room dimensions, row configurations, fans that are readily available, extensable, and separates concerns so that algorithms and learning models can be applied to find control optimizations that can generalize to different data center environments.
+
+Perspective users
+=================
+
+Mechanical engineers and algorithm engineers aiming to improve energy efficiency in data centers.
+
+Users can use the existing environment implementations or create their own implementations from the examples to test new (learning) algorithms on data center environments and find optimal solutions to reduce energy consumption.
+
+Mechanical engineers can focus on the parametrization and validity of the models and algorithm engineers can focus on the design, implementation, profiling and optimization of control algorithms.
+
+System architecture
+===================
+
+.. image:: ./opendce_diagrams.png
+
+API description
+===============
+
+The `OpenAI Gym <https://github.com/openai/gym>`_ API will server as a guideline to the API found in this project.
```

Comment:
It's good to have a reference for API design.  It will be valuable to provide analysis for why the Gym API is used as a model.